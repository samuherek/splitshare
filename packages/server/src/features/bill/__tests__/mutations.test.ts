import gql from 'graphql-tag';
import createTestServer from '../../../test-utils/server';

describe('bill mutations', () => {
  test('createBill: mutation should create and return a created bill', async () => {
    const CREATE_BILL = gql`
      mutation createBill($input: CreateBillInput!) {
        createBill(input: $input) {
          id
          name
          icon
          createdAt
          updatedAt
        }
      }
    `;

    const { mutate } = await createTestServer({
      user: { id: 1 },
      models: {
        Bill: {
          createOne: jest.fn(() => ({
            id: 1,
            name: 'new bill',
            icon: null,
            createdAt: new Date('2000-01-01'),
            updatedAt: new Date('2000-01-01'),
          })),
        },
      },
    });

    const res = await mutate({
      mutation: CREATE_BILL,
      variables: { input: { name: 'new bill' } },
    });

    expect(res).toMatchSnapshot();
  });
});
