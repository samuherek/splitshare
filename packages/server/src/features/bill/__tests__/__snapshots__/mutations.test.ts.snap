// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bill mutations createBill: mutation should create and return a created bill 1`] = `
Object {
  "data": Object {
    "createBill": Object {
      "createdAt": "2000-01-01",
      "icon": null,
      "id": "1",
      "name": "new bill",
      "updatedAt": "2000-01-01T00:00:00.000Z",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
